src/Microsoft.OData.Core/ODataMessageWriterSettings.cs

```diff
@@ -423,6 +437,11 @@ private void CopyFrom(ODataMessageWriterSettings other)
             this.UseKeyAsSegment = other.UseKeyAsSegment;
             this.useFormat = other.useFormat;
             this.Version = other.Version;
+            this.validations = other.validations;
+            this.BasicValidation = other.BasicValidation;
+            this.ThrowOnDuplicatePropertyNames = other.ThrowOnDuplicatePropertyNames;
+            this.ThrowOnNullValuesForNonNullablePrimitiveTypes = other.ThrowOnNullValuesForNonNullablePrimitiveTypes;
+            this.ThrowOnUndeclaredProperty = other.ThrowOnUndeclaredProperty;

```

Comment:
By `this.validations = other.validations`, I'm copying the field directly instead of calling the property accessors. Note that difference between `validations` and `Validations`.